### PR TITLE
#17539 fix error when using `batch()` with `indexBy()` with MSSQL

### PR DESCRIPTION
**Pull request by alexkart**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #17539

### CODE REVIEW

**Comment by alexkart** [comment]
Is everything ok?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Yes. Looks good to me but I'd like @yiisoft/reviewers to check it as well.

File: framework/db/BatchQueryResult.php [hunk 1]
```diff
@@ -214,8 +214,11 @@ private function getDbDriverName()
             return $this->db->driverName;
         }
 
-        if (isset($this->_batch[0]->db->driverName)) {
-            return $this->_batch[0]->db->driverName;
+        if (isset($this->_batch) && is_array($this->_batch) && !empty($this->_batch)) {
```
Comment (rob006) [inline framework/db/BatchQueryResult.php]: `isset($this->_batch)` is redundant (`!empty($this->_batch)` already ensures that). `is_array($this->_batch)` probably too - `$this->_batch` is either `null` or `array`, so simple `!empty($this->_batch)` should be enough.

File: tests/framework/db/BatchQueryResultTest.php [hunk 2]
```diff
@@ -118,6 +118,16 @@ public function testBatchWithoutDbParameter()
         $this->assertEquals('user3', $customers[2]->name);
     }
 
+    public function testBatchWithIndexBy()
+    {
+        $query = Customer::find()->orderBy('id')->limit(3)->indexBy('id');
+        $customers = $this->getAllRowsFromBach($query->batch(2));
```
Comment (rob006) [inline tests/framework/db/BatchQueryResultTest.php]: There is a typo in this method name - it should be `getAllRowsFromBatch()`.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**